Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,588 @@
+          .setLabel(_('Visible on start'))
+      );
+
+      return objectProperties;
+    };
+    objectBBText.setRawJSONContent(
+      JSON.stringify({
+        text:
+          '[b]bold[/b] [i]italic[/i] [size=15]smaller[/size] [font=times]times[/font] font\n[spacing=12]spaced out[/spacing]\n[outline=yellow]outlined[/outline] [shadow=red]DropShadow[/shadow] ',
+        opacity: 255,
+        fontSize: '20',
+        visible: true,
+        color: '#000000',
+        fontFamily: 'Arial',
+        align: 'left',
+        wordWrap: true,
+      })
+    );
+
+    objectBBText.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    objectBBText.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'BBText',
+        _('BBText'),
+        _(
+          'Displays a rich text label using BBCode markup (allowing to set parts of the text as bold, italic, use different colors and shadows).'
+        ),
+        'JsPlatform/Extensions/bbcode32.png',
+        objectBBText
+      )
+      .setIncludeFile('Extensions/BBText/bbtextruntimeobject.js')
+      .addIncludeFile('Extensions/BBText/bbtextruntimeobject-pixi-renderer.js')
+      .addIncludeFile(
+        'Extensions/BBText/pixi-multistyle-text/dist/pixi-multistyle-text.umd.js'
+      );
+
+    // Utility function to add both a setter and a getter to a property from a list. Useful for setting multiple generic properties
+    const addSettersAndGettersToObjectHelper = (
+      gdObject,
+      properties,
+      objectName
+    ) => {
+      properties.forEach(property => {
+        const parameterType =
+          property.type === 'boolean' ? 'yesorno' : property.type;
+
+        // Add all the generic GETTERS
+        if (parameterType === 'number') {
+          gdObject
+            .addExpression(
+              `Get${property.functionName}`,
+              _('Get ') + property.description,
+              _('Get ') + property.description,
+              _(''),
+              '',
+              `${property.iconPath}.png`,
+              `${property.iconPath}.png`
+            )
+            .addParameter(
+              'object',
+              _(`${objectName} object`),
+              objectName,
+              false
+            )
+            .getCodeExtraInformation()
+            .setFunctionName(`get${property.functionName}`);
+        } else {
+          gdObject
+            .addStrExpression(
+              `Get${property.functionName}`,
+              _('Get ') + property.description,
+              _('Get ') + property.description,
+              _(''),
+              '',
+              `${property.iconPath}.png`,
```

Comment:
Adding ".png" automagically seems an implicit addition that is hard to discover without reading the code - and suprising considering that you named the property "iconPath", which implies a path to a file. Better have iconPath to include the extension. 
Explicit is better than implicit, even if it's for a few more ".png" to add when calling the function.

Reply:
(here and everywhere iconPath is used)